src/components/annotations/common_defaults.js

```diff
@@ -35,11 +35,18 @@ module.exports = function handleAnnotationCommonDefaults(annIn, annOut, fullLayo
     if(h) coerce('valign');
 
     if(showArrow) {
+        var arrowside = coerce('arrowside');
+        var arrowhead = coerce('arrowhead');
+        var arrowsize = coerce('arrowsize');
+
+        if(arrowside.indexOf('start') !== -1) {
+            coerce('startarrowhead', arrowhead);
```

Comment:
```
coerce('startarrowsize', arrowsize);
```

and

```
coerce('startstandoff');
```

should be inside this block as well - as they have no effect w/o start arrow.

Reply:
There was possibility to use `standoff` with `arrowhead: 0` (no arrow) before my changes and I decided to keep this functionality for ‘startstandoff’ -> when we have `arrowside: none`, we still can set both properties of standoff and see the result. 
Please just confirm that it is not expected behaviour and I will apply suggested changes.

Reply:
Ha good point. I think you have it right here.

Reply:
Right, so `(start)standoff` always needs to be coerced - but doesn't `startarrowsize` go in here?

Also if we have no `'end'` we don't need `arrowhead` and `arrowsize` do we?

Reply:
if we agree to have `(start)standoff` without arrow heads, then we always need coerced `startarrowsize`, `arrowhead` and `arrowsize`  because `(start)standoff` depends on it.

See lines from 40 to 47 in draw_arrow_head.js
https://github.com/plotly/plotly.js/pull/2164/files#diff-7f4a16b3f6ab73b9262d71404b2d920eR40

Reply:
Seems like @apalchys posted and deleted a comment, so maybe he noticed what I'm about to say... (incidentally, if that's the case, let me encourage you to edit comments, using ~~strikethrough~~ if necessary, rather than deleting them, as it makes the conversation flow clearer)

`(start)arrowhead` and `(start)arrowsize` are not needed *logically* when we're not showing a head on that end, so that's a warning sign that providing them could lead to bugs. Indeed, I think there is a subtle one: if you choose an arrowhead that has a `backoff` (including the default `arrowhead: 1`! `backoff` is how much you need to move the end of the line backward so the arrowhead can point to the right place without the line jutting through it) but you don't show that arrowhead, you'll get an excessive standoff.

`options.arrowhead || 0` ensures an un-coerced `arrowhead` is treated as no arrow, and we may need `options.arrowsize || 0` (this value probably doesn't matter as long as it's numeric) to make the math well-defined.

Reply:
sorry,  I recognized that my answer is wrong right after I submit it. I will commit an update soon.

Reply:
Please check my commit - https://github.com/plotly/plotly.js/pull/2164/commits/191d5d2055ba787195cea9d6fb936a806b800490
Is it what you expected?

Reply:
Perfect, thanks! Looks like this tweaked one more test image (`gl2d_annotations`) but once that's fixed I think we're ready to go 🎉